src/Domain/Tags/Faces.hpp

```diff
@@ -0,0 +1,119 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+
+namespace domain {
+namespace Tags {
+
+/// The `Tag` on element faces
+template <size_t Dim, typename Tag>
+struct Faces : db::PrefixTag, db::SimpleTag {
+  using tag = Tag;
+  using type = DirectionMap<Dim, typename Tag::type>;
+};
+
+}  // namespace Tags
+
+namespace FacesTag_detail {
+template <typename T, typename = std::void_t<>>
+struct get_volume_tags_impl {
+  using type = tmpl::list<>;
+};
+template <typename T>
+struct get_volume_tags_impl<T, std::void_t<typename T::volume_tags>> {
+  using type = typename T::volume_tags;
+};
+}  // namespace FacesTag_detail
+
+/// Retrieve the `T::volume_tags`, or `tmpl::list<>` if `T` doesn't have the
+/// type alias
+template <typename T>
+using get_volume_tags =
+    tmpl::type_from<FacesTag_detail::get_volume_tags_impl<T>>;
```

Comment:
Just use `::type`.  It's more standard for metafunctions.  (We generally use `type_from` to extract from a `tmpl::type_`.)